annotations/manifest_upload.py

```diff
@@ -71,13 +71,14 @@ def edit_manifest(file_path, col_types_dict):
     # Adjust data types to match table schema
     for columnName in df:
         if col_types_dict[columnName] == list:
+            df[columnName] = df[columnName].astype(str)
             df[columnName] = df[columnName].str.split(', ')
         else:
             for k, v in col_types_dict.items():
                 df[columnName] = df[columnName].astype(
                     col_types_dict[columnName])
             # For columns with USERID as datatype, remove .0 tacked on in data type conversion.
-            if col_types_dict[columnName] == str:
+            if col_types_dict.get(columnName) == 'USERID':
                 df[columnName] = df[columnName].str.rstrip('.0')
```

Comment:
Whoops!  I just learned that `rstrip()` reads the argument as a _list of characters_, **not** a substring.  That is:

```
s.rstrip(".0")
```

will remove `.` **or** `0` if they're found at the end of the string, NOT `.0` like I thought.  My bad!

For robustness, let's not use rstrip() at all (in case a user ID has a `0` at the end, which would then be stripped too).  Instead, let's use pandas' `replace()`:

```suggestion
                df[columnName] = df[columnName].replace("\.0$", "")
```